### PR TITLE
add service account perms for agent

### PR DESCRIPTION
**Pull request by bcsherma**
## Description

What does the pull request do? If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Ticket

Does this PR fix an existing issue? If yes, provide a link to the ticket here:

## Checklist

Check if your PR fulfills the following requirements. Put an `X` in the boxes that apply.

- [ ] Files I edited were previewed on my local development server with `yarn start`. My changes did not break the local preview.
- [ ] Build (`yarn docusaurus build`) was run locally and successfully without errors or warnings.
- [ ] I merged the latest changes from `main` into my feature branch before submitting this PR.

### CODE REVIEW

**Review by ngrayluna — APPROVED** [review]

File: docs/guides/launch/run-agent.md [hunk 1]
```diff
@@ -62,7 +61,7 @@ builder:
   type: docker|kaniko|noop
 ```
 
-The `environment`, `registry`, and `builder` keys are optional. By default, the agent will use no cloud environment, a local docker registry, and a docker builder. 
+The `environment`, `registry`, and `builder` keys are optional. By default, the agent will use no cloud environment, a local docker registry, and a docker builder.
```
Comment (tristan-spaulding2) [inline docs/guides/launch/run-agent.md]: nitpick: would be nice to consistently capitalize Docker

File: docs/guides/launch/run-agent.md [hunk 2]
```diff
@@ -208,6 +206,148 @@ builder:
...
+  ]
+}
+
+```
+
+</TabItem>
+
+<TabItem value="gcp">
+
+The gcp credentials in the agents environment must include the following permissions in order to access various services:
```
Comment (tristan-spaulding2) [inline docs/guides/launch/run-agent.md:353]: nitpick:capitalization

**Review by tristan-spaulding2 — APPROVED** [review]

File: docs/guides/launch/kubernetes.md [hunk 3]
```diff
@@ -11,12 +11,12 @@ This guide demonstrates how to use W&B Launch to run ML workloads on a kubernete
 The launch agent uses [Kaniko](https://github.com/GoogleContainerTools/kaniko) to build container images inside of k8s. Kaniko is a tool to build container images from a Dockerfile, inside a container or k8s cluster. Kaniko doesn’t depend on a Docker daemon and executes each command within a Dockerfile completely in userspace. This enables building container images in environments that can’t easily or securely run a Docker daemon, such as a standard k8s cluster.
 
 :::tip
+
```
Comment (KyleGoyette) [inline docs/guides/launch/kubernetes.md:14]: errant new line?

File: docs/guides/launch/run-agent.md [hunk 4]
```diff
@@ -172,24 +170,24 @@ builder:
...
 
-If you are running your own k8s cluster, you will need to create a k8s secret that contains the credentials for your cloud environment. To grant access to GCP, this secret should contain a [service account json](https://cloud.google.com/iam/docs/keys-create-delete#creating). To grant access to AWS, this secret should contain an [AWS credentials file](https://docs.aws.amazon.com/sdk-for-php/v3/developer-guide/guide_credentials_profiles.html). You can configure the Kaniko builder to use this secret by setting the following keys in your builder config:
+If you are running your own Kubernetes cluster, you will need to create a Kubernetes secret that contains the credentials for your cloud environemnt. To grant access to GCP, this secret should contain a [service account json](https://cloud.google.com/iam/docs/keys-create-delete#creating). To grant access to AWS, this secret should contain an [AWS credentials file](https://docs.aws.amazon.com/sdk-for-php/v3/developer-guide/guide_credentials_profiles.html). You can configure the Kaniko builder to use this secret by setting the following keys in your builder config:
```
Comment (KyleGoyette) [inline docs/guides/launch/run-agent.md]: `environemnt` is spelled wrong

File: docs/guides/launch/run-agent.md [hunk 5]
```diff
@@ -208,6 +206,148 @@ builder:
...
+      "Action": "iam:PassRole",
+      "Resource": "<ARN-OF-ROLE-TO-PASS>"
+    },
+    {
+      "Effect": "Allow",
+      "Action": "kms:CreateGrant",
+      "Resource": "<ARN-OF-KMS-KEY>",
+      "Condition": {
+        "StringEquals": {
+          "kms:ViaService": "sagemaker.<REGION>.amazonaws.com",
```
Comment (KyleGoyette) [inline docs/guides/launch/run-agent.md:335]: @bcsherma did you add the creategrant and passrole requirements? Can we internally document why we need these?

File: docs/guides/launch/run-agent.md [hunk 6]
```diff
@@ -172,24 +170,24 @@ builder:
 
 <TabItem value="kaniko">
 
-Set `type: kaniko` and configure either a GCP or AWS environment to use Kaniko to build container images in Kubernetes. The `kaniko` builder requires the `build-context-store` key to be set. The `build-context-store` key should be an S3 or GCS prefix that the agent will use to store build contexts, depending on the environment that is configured. The `build-job-name` can be used to specify a name prefix for the k8s job that the agent will use to build images.
+Set `type: kaniko` and configure either a GCP or AWS environment to use Kaniko to build container images in Kubernetes. The `kaniko` builder requires the `build-context-store` key to be set. The `build-context-store` key should be an S3 or GCS prefix that the agent will use to store build contexts, depending on the environment that is configured. The `build-job-name` can be used to specify a name prefix for the Kubernetes job that the agent will use to build images.
```
Comment (KyleGoyette) [inline docs/guides/launch/run-agent.md]: What is the build context store in Azure?

**Review by KyleGoyette — COMMENTED** [review]
High value changes here!

A few things:
1. Seems like some of this PR adds Azure, but some doesn't, let's make sure we're consistent there.
2. (Non blocking) There's some perms I've not seen it AWS for Sagemaker, I would love to know more about these. 
3. Some typos and capitalizations as @tristan-spaulding2 pointed out

File: docs/guides/launch/kubernetes.md (hunk 3 above)
Comment (bcsherma) [reply docs/guides/launch/kubernetes.md:14]: Actually having newlines around lists is apparently a markdown style convention that gets linted. It has no effect on the display.

File: docs/guides/launch/run-agent.md (hunk 6 above)
Comment (bcsherma) [reply docs/guides/launch/run-agent.md]: added

File: docs/guides/launch/run-agent.md (hunk 5 above)
Comment (bcsherma) [reply docs/guides/launch/run-agent.md:335]: I got this from sagemaker docs: https://docs.aws.amazon.com/sagemaker/latest/dg/api-permissions-reference.html#:~:text=CreateTrainingJob,permits%20this%20action

Passrole is needed for the agent to give the training job it creates permission to assume the sagemaker execution role.

CreateGrant is only needed in certain conditions. We could drop that...?

File: docs/guides/launch/run-agent.md (hunk 4 above)
Comment (bcsherma) [reply docs/guides/launch/run-agent.md]: I think this got fixed in antoehr commit I don't see any typos in environment

File: docs/guides/launch/run-agent.md (hunk 5 above)
Comment (KyleGoyette) [reply docs/guides/launch/run-agent.md:335]: We should probably just document the CreateGrant conditions. I wouldn't block on this

**Review by KyleGoyette — APPROVED** [review]

**Pull request merged**